src/crimson/os/seastore/segment_manager/zns.cc

```diff
@@ -24,66 +21,63 @@ static open_device_ret open_device(
   const std::string &path,
   seastar::open_flags mode)
 {
+  LOG_PREFIX(ZNSSegmentManager::open_device);
   return seastar::file_stat(
     path, seastar::follow_symlink::yes
-  ).then([mode, &path](auto stat) mutable{
+  ).then([FNAME, mode, &path](auto stat) mutable{
     return seastar::open_file_dma(path, mode).then([=](auto file){
-      logger().error(
-	"open_device: open successful, size {}",
-	stat.size);
+       DEBUG("open of device {} successful, size {}",
+        path,
+        stat.size);
       return std::make_pair(file, stat);
     });
   }).handle_exception(
-    [](auto e) -> open_device_ret {
-      logger().error(
-	"open_device: got error {}",
+    [FNAME](auto e) -> open_device_ret {
+	ERROR("got error {}",
 	e);
       return crimson::ct_error::input_output_error::make();
     }
   );
 }
 
 static zns_sm_metadata_t make_metadata(
+  uint64_t size,
   seastore_meta_t meta,
   const seastar::stat_data &data,
   size_t zone_size,
   size_t zone_capacity,
   size_t num_zones)
 {
-  using crimson::common::get_conf;
-  
-  auto config_size = get_conf<Option::size_t>(
-    "seastore_device_size");
```

Comment:
@athanatos can we ignore this configuration, or need to treat it as the maximum size when the real device size is larger?

Reply:
@aravind-wdc if decide to ignore this configuration, please add a warn log to notify user.

Reply:
We could also allow zone_size == 0 and in that case, read it from config file.
Then will need to remove the assert() for zone_size == 0.
What do you think would be a better approach ? 
Adding WARN log and ignoring the config  
Or 
Ignoring zone_size == 0 situation and use config values in that case.

Reply:
> We could also allow zone_size == 0

zone_size is from BLKGETZONESZ, which is impossible to be 0, right?

I think adding a WARN log should be OK for now.

Reply:
Added  WARN log.